src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -45,21 +46,20 @@ public static Encoding OutputEncoding
             get { return GetConsoleEncoding(); }
         }
 
-        private static SyncTextReader s_stdInReader;
+        private static StdInReader s_stdInReader;
         private const int DefaultBufferSize = 255;
 
-        private static SyncTextReader StdInReader
+        private static StdInReader StdInReader
         {
             get
             {
                 EnsureInitialized();
 
                 return Console.EnsureInitialized(
                         ref s_stdInReader,
-                        () => SyncTextReader.GetSynchronizedTextReader(
-                            new StdInReader(
-                                encoding: new ConsoleEncoding(Console.InputEncoding), // This ensures no prefix is written to the stream.
-                                bufferSize: DefaultBufferSize)));
+                        () => new StdInReader(
```

Comment:
As far as I can tell from code review, this breaks thread-safety.  Multiple threads can be doing `Console.ReadLine()` for example, and each should behave atomically with respect to the other.  That atomicity was provided by the SynchronizedTextReader.  But now, the StdInReader property is just returning an unadorned StdInReader, such that calls to Console.ReadLine, for example, will just in turn call to StdInReader.ReadLine, which is not thread-safe.

Reply:
I believe I fixed it.

Reply:
Yes, thanks.